subprojects/core/src/main/java/org/gradle/util/RetryUtil.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.util;
+
+import com.google.common.base.Function;
+import com.google.common.base.Throwables;
+import groovy.lang.Closure;
+
+public final class RetryUtil {
```

Comment:
I'd move this class into the `internal-testing` module where we store testing infrastructure for unit tests.